scripts/tests/test_zkai_attention_kv_d32_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,247 @@
+            ("sidecar envelope", gate.SIDECAR_ENVELOPE_JSON, gate.SIDECAR_ENVELOPE_SIZE_BYTES),
+            ("fused envelope", gate.FUSED_ENVELOPE_JSON, gate.FUSED_ENVELOPE_SIZE_BYTES),
+        )
+        for label, path, expected_size in cases:
+            raw = path.read_bytes()
+            self.assertEqual(len(raw), expected_size)
+            with self.assertRaisesRegex(gate.AttentionKvD32FusedSoftmaxTableGateError, f"{label} size drift"):
+                gate.expect_artifact_size(raw + b" ", expected_size, label)
+
+    def test_write_json_and_tsv_round_trip(self):
+        with tempfile.TemporaryDirectory() as tmp:
+            tmp = gate.pathlib.Path(tmp)
```

Ruff (0.15.12)

[warning] `with` statement variable `tmp` overwritten by assignment target

(PLW2901)